### PR TITLE
Update Tendermint version to 0.34.x

### PR DESCRIPTION
**Pull request by hacheigriega**

### CODE REVIEW

**Comment by yoongbok-lee** [comment]
We can upstream this to burrow too after we're done 👍  thanks for your work!

**Comment by hacheigriega** [comment]
> We can upstream this to burrow too after we're done 👍 thanks for your work!

No problem. Sounds good!

**Comment by yoongbok-lee** [comment]
BTW, we're referencing v0.1.0 release from shentu repo, so you might want to rebase to v0.1.0 before making any further changes (v0.1.0 corresponds to whatever we were using + some native eWASM contracts)

File: consensus/tendermint/sign_info.go [hunk 1]
```diff
@@ -37,7 +41,7 @@ func voteToStep(vote *types.Vote) int8 {
 type LastSignedInfo struct {
 	sync.Mutex
 	Height    int64           `json:"height"`
-	Round     int             `json:"round"`
+	Round     int32           `json:"round"`
```
Comment (yoongbok-lee) [inline consensus/tendermint/sign_info.go:44]: 32 or should it be 64?

File: crypto/tendermint.go [hunk 2]
```diff
@@ -18,51 +20,102 @@ func NodeIDFromAddress(id Address) p2p.ID {
...
+// 		}, nil
+// 	case CurveTypeSecp256k1.ABCIType():
+// 		return PublicKey{
+// 			CurveType: CurveTypeSecp256k1,
+// 			PublicKey: pubKey.Data,
+// 		}, nil
+// 	}
+// 	return PublicKey{}, fmt.Errorf("did not recognise ABCI PubKey type: %s", pubKey.Type)
+// }
+
```
Comment (yoongbok-lee) [inline crypto/tendermint.go:73]: Should delete any unused codes

File: encoding/protobuf.go [hunk 3]
```diff
@@ -5,7 +5,8 @@ import (
 	"fmt"
 	"io"
 
-	"github.com/golang/protobuf/proto"
+	//"github.com/golang/protobuf/proto"
```
Comment (yoongbok-lee) [inline encoding/protobuf.go:8]: should delete any commented our codes

**Review by yoongbok-lee — APPROVED** [review]
LGTM

File: consensus/tendermint/sign_info.go (hunk 1 above)
Comment (hacheigriega) [reply consensus/tendermint/sign_info.go:44]: Tendermint chose 32 to reduce proto sizes https://github.com/tendermint/tendermint/pull/4582

File: crypto/tendermint.go (hunk 2 above)
Comment (hacheigriega) [reply crypto/tendermint.go:73]: fixed.

File: encoding/protobuf.go (hunk 3 above)
Comment (hacheigriega) [reply encoding/protobuf.go:8]: fixed. thanks!

**Pull request merged**